spec/syntax.md

```diff
@@ -705,6 +705,18 @@ on the pairing, ordering, or contents of _markup_ during _formatting_.
 Examples in this section are meant to be illustrative and
 might not match future requirements or usage.
 
+During the specification's "tech preview" phase,
+implementations SHOULD NOT implement functionality using function or markup options
+that might later be replaced by attributes or some other solution.
+This includes:
+- Specifying the locale of an expression or markup part that is different from the message's overall locale.
+- Explicitly linking opening and closing markup parts using option values,
+  such as a `startRef` on a closing markup part matching the `id` or `ref` value of an opening markup part.
+- Defining a namespace of options that have no formatting runtime impact,
+  and are used to communicate information to translators or translation tools.
+
+The above restrictions will be reviewed and removed before the end of the tech preview.
```

Comment:
I would really prefer the normative SHOULD NOT, because that's a less handwavy way to say the same as "are cautioned to avoid ...", but I could live with the more verbose note as well.

Is there a reason to leave this out, though?
> - Defining a namespace of options that have no formatting runtime impact,
>   and are used to communicate information to translators or translation tools.